build-tools/automation/azure-pipelines.yaml

```diff
@@ -0,0 +1,213 @@
+      condition: always()
+
+    - task: PublishPipelineArtifact@0
+      displayName: upload artifacts
+      inputs:
+        artifactName: mac-build-results
+        targetPath: $(Build.ArtifactStagingDirectory)
+      condition: always()
+
+- stage: win_build_test
+  displayName: Win
+  dependsOn: prepare
+  jobs:
+  - job: build_test
+    displayName: Build and Test
+    pool: VSEng-MicroBuildVS2017
+    timeoutInMinutes: 360
+    cancelTimeoutInMinutes: 5
+    steps:
+    - task: DownloadPipelineArtifact@1
+      inputs:
+        artifactName: $(BundleArtifactName)
+        downloadPath: $(System.DefaultWorkingDirectory)
+
+    - task: MSBuild@1
+      displayName: msbuild Xamarin.Android /t:Prepare
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:Prepare /bl:bin\Build$(Configuration)\msbuild-prepare.binlog /p:BundleRootPath=$(System.DefaultWorkingDirectory)
+
+    - task: MSBuild@1
+      displayName: msbuild Xamarin.Android /t:Build
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:Build /bl:bin\Build$(Configuration)\msbuild-build.binlog /p:BundleRootPath=$(System.DefaultWorkingDirectory)
+
+    - task: MSBuild@1
+      displayName: msbuild create-vsix
+      inputs:
+        solution: build-tools\create-vsix\create-vsix.csproj
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /p:CreateVsixContainer=True /p:ZipPackageCompressionLevel=Normal /bl:bin\Build$(Configuration)\msbuild-create-vsix.binlog
+
+    - task: CmdLine@1
+      displayName: xabuild Xamarin.Android-Tests
+      inputs:
+        filename: bin\$(XA.Build.Configuration)\bin\xabuild.exe
+        arguments: Xamarin.Android-Tests.sln /p:Configuration=$(XA.Build.Configuration) /p:XAIntegratedTests=False /bl:bin\Build$(Configuration)\msbuild-build-tests.binlog
+
+    - task: MSBuild@1
+      displayName: nunit Xamarin.Android.Build.Tests
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:RunNUnitTests /bl:bin\Build$(Configuration)\msbuild-run-nunit-tests.binlog
+      timeoutInMinutes: 240
+
+    - task: MSBuild@1
+      displayName: nunit Java.Interop Tests
+      inputs:
+        solution: Xamarin.Android.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: >
+          /t:RunJavaInteropTests
+          /p:TestAssembly="bin\Test$(XA.Build.Configuration)\generator-Tests.dll;bin\Test$(XA.Build.Configuration)\Java.Interop.Tools.JavaCallableWrappers-Tests.dll;bin\Test$(XA.Build.Configuration)\LogcatParse-Tests.dll;bin\Test$(XA.Build.Configuration)\Xamarin.Android.Tools.ApiXmlAdjuster-Tests.dll;bin\Test$(XA.Build.Configuration)\Xamarin.Android.Tools.Bytecode-Tests.dll"
+          /bl:bin\Build$(Configuration)\msbuild-run-ji-tests.binlog
```

Comment:
Unit test execution output should go into `bin/Test$(Configuration)`, not `bin/Build$(Configuration)`.

Also, shouldn't this be `$(XA.Build.Configuration)`, not `$(Configuration)`?

Should we instead introduce a high-level/global `$(Configuration)` variable which is set to `$(XA.Build.Configuration)`, and use the shorter `$(Configuration)` everywhere?

Reply:
This is indeed a typo left over from some refactoring.

I didn't want to introduce a global variable for config named `Configuration`, as we likely will want to use different configurations in different steps in different stages (especially WRT testing). MSBuild will pick up an environment variable named `Configuration` and use it automatically, and I wanted to avoid the possibility of a global Configuration value being used behind the scenes in places where it was not otherwise explicitly set.